pkg/collect/collect.go

```diff
@@ -192,3 +192,22 @@ func CollectRemote(c *troubleshootv1beta2.RemoteCollector, additionalRedactors *
 	collectResult.AllCollectedData = allCollectedData
 	return collectResult, nil
 }
+
+func DedupHostCollectors(allCollectors []*troubleshootv1beta2.HostCollect) []*troubleshootv1beta2.HostCollect {
```

Comment:
I would DRY this code. `DedupHostCollectors` and `DedupHostAnalyzers` are very similar and can be one function. Think go Generics

Reply:
Thanks @banjoh!

The reason I don't why a generic function is that there're existing dedupe functions

https://github.com/replicatedhq/troubleshoot/blob/6b368f22213786d25bf71172d587325c19fbc7ec/pkg/collect/collector.go#L266

https://github.com/replicatedhq/troubleshoot/blob/db871e6889f69856be3084230569d1e558e36613/pkg/analyze/analyzer.go#L261

I think a separate PR is needed to have a generic function and replace others dedupe functions as well.

Reply:
I was referring to https://gobyexample.com/generics, but it can be done on another PR

Reply:
Specifically, like so

```go
func Dedup[T any](allAnalyzers []T) []T {
	seen := make(map[string]bool)
	out := []T{}

	for _, analyzer := range allAnalyzers {
		data, err := json.Marshal(analyzer)
		if err != nil {
			out = append(out, analyzer)
			continue
		}
		key := string(data)
		if _, ok := seen[key]; !ok {
			out = append(out, analyzer)
			seen[key] = true
		}
	}
	return out
}
```

Then use it like this

```go
func DedupHostCollectors(allCollectors []*troubleshootv1beta2.HostCollect) []*troubleshootv1beta2.HostCollect {
	return utils.Dedup(allCollectors)
}

func DedupHostAnalyzers(allAnalyzers []*troubleshootv1beta2.HostAnalyze) []*troubleshootv1beta2.HostAnalyze {
	return interfaceutils.Dedup(allAnalyzers)
}
```

You probably do not even need `DedupHostAnalyzers` and `DedupHostCollectors`

Reply:
many thanks @banjoh! with feedback from @DexterYan  as well on the collector dedup, I have followed our suggestions and updated accordingly.